### PR TITLE
Add iOS source context info to RN debug symbols page

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

related to https://github.com/getsentry/sentry-react-native/issues/2748

### CODE REVIEW

File: src/platforms/react-native/upload-debug.mdx [hunk 1]
```diff
@@ -16,6 +16,10 @@ Errors raised from the native layer in Android apps require certain debug inform
 
 Upload the [Android ProGuard/R8 mapping files and native symbols](/platforms/android/proguard/) by the recommended method of using our Gradle integration.
 
+## Uploading With Xcode
+
+Build phase [`Upload Debug Symbols to Sentry`](/platforms/react-native/manual-setup/manual-setup/#upload-debug-symbols-to-sentry) uses `sentry-cli` to upload native iOS debug symbols. To enable [Source Context](/platforms/apple/guides/ios/data-management/debug-files/source-context/) you can use `SENTRY_INCLUDE_NATIVE_SOURCES=true` environmental variable set before build.
```
Comment (kahest) [inline src/platforms/react-native/upload-debug.mdx:21]:
```suggestion
Build phase [`Upload Debug Symbols to Sentry`](/platforms/react-native/manual-setup/manual-setup/#upload-debug-symbols-to-sentry) uses `sentry-cli` to upload native iOS debug symbols. To enable [Source Context](/platforms/apple/guides/ios/data-management/debug-files/source-context/), set `SENTRY_INCLUDE_NATIVE_SOURCES=true` environmental before build.
```

Comment (kahest) [reply src/platforms/react-native/upload-debug.mdx:21]: `l`: maybe also link to https://docs.sentry.io/platforms/apple/dsym/#sentry-cli

**Review by kahest — APPROVED** [review]
LGTM, just tiny suggestions

File: src/platforms/react-native/upload-debug.mdx (hunk 1 above)
Comment (shanamatthews) [inline src/platforms/react-native/upload-debug.mdx:21]:
```suggestion
To upload native iOS debug symbols, set up a Run Script build phase for uploading debug symbols to Sentry with the Sentry CLI. You can find documentation for this on the [Manual Configuration](/platforms/react-native/manual-setup/manual-setup/#upload-debug-symbols-to-sentry) page.

To enable [Source Context](/platforms/apple/guides/ios/data-management/debug-files/source-context/) set `SENTRY_INCLUDE_NATIVE_SOURCES=true` before the build.
```

Comment (shanamatthews) [reply src/platforms/react-native/upload-debug.mdx:21]: Alternative phrasing to put the user's objective first.

**Review by shanamatthews — APPROVED** [review]

File: src/platforms/react-native/upload-debug.mdx (hunk 1 above)
Comment (krystofwoldrich) [reply src/platforms/react-native/upload-debug.mdx]: Maybe we can link it right away in the manual setup docs.
https://github.com/getsentry/sentry-docs/pull/7151

**Pull request merged**